CHANGELOG.md

```diff
@@ -4,6 +4,7 @@
 
 - feat: Expose SentryDebugImageProvider (#1094)
 - docs: Improve code doc on start and endSession (#1098)
+- remove libz (#1109)
```

Comment:
```suggestion
- ref: Remove libz (#1109)
```

Reply:
I honestly can't tell, this will boil down to us testing this PR on all platforms using gzip compression to make sure this doesn't break it.

Reply:
GitHub Actions don't offer testing on all our supported platforms for macOS; see https://github.com/actions/virtual-environments/tree/main/images/macos. So maybe we keep this @bruno-garcia and instead add a note somewhere that we rather not remove this, because we are not sure if it breaks somewhere.

Reply:
Not remove it because we don't know if it breaks is suboptimal IHMO.
I'll do more device testing then but I rather we remove this if it's not needed.
Ideally we'd validate this in CI even, so we can be confident we can remove an unused dependency without fear of breaking things

Reply:
What about we bump the min version macOS 10.11?

Reply:
Even when bumping to 10.11, we should validate this fix.